### PR TITLE
Upgrade to commander 7

### PR DESCRIPTION
**Pull request by lydell**
This allows getting rid of all workarounds we had for commander.

### CODE REVIEW

File: lib/elm-test.js [hunk 1]
```diff
@@ -150,23 +98,24 @@ function main() {
...
     )
-    .option(
-      '--seed <int>',
-      'Run with a previous fuzzer seed',
-      parsePositiveInteger('--seed <int>'),
-      Math.floor(Math.random() * 407199254740991) + 1000
+    .addOption(
+      new Option('--seed <int>', 'Run with a previous fuzzer seed')
+        .default(Math.floor(Math.random() * 407199254740991) + 1000, 'random')
+        .argParser(parsePositiveInteger)
```
Comment (harrysarson) [inline lib/elm-test.js:104]: Out of interest what is going here? The `'random'` looks a bit special.

File: lib/elm-test.js [hunk 2]
```diff
@@ -234,7 +179,10 @@ function main() {
       const make = async () => {
         await Generate.generateElmJson(project, () => null);
         await Compile.compileSources(
-          FindTests.resolveGlobs(testFileGlobs, project.rootDir),
+          FindTests.resolveGlobs(
+            testFileGlobs.length === 0 ? [project.testsDir] : testFileGlobs,
+            project.rootDir
+          ),
```
Comment (harrysarson) [inline lib/elm-test.js:185]: Could we add a unit test?

**Review by harrysarson — APPROVED** [review]
Looks good (with a regression test)! Thanks :)

File: lib/elm-test.js (hunk 1 above)
Comment (lydell) [reply lib/elm-test.js:104]: Now, the helps looks like:

```
  --seed <int>       Run with a previous fuzzer seed (default: random)
```

Previously, it was:

```
  --seed <int>                   Run with a previous fuzzer seed (default: 390695695167635)
```

With a different number each time.

So it wasn’t clear if the default seed was a constant or what it was.

File: lib/elm-test.js (hunk 2 above)
Comment (lydell) [reply lib/elm-test.js:185]: Added! :+1:

File: lib/elm-test.js (hunk 1 above)
Comment (harrysarson) [reply lib/elm-test.js:104]: Nice!

**Comment by harrysarson** [comment]
Does the elm make test fail without the fix? And does the readme need updating? Otherwise we are good to merge :)

**Comment by lydell** [comment]
> Does the elm make test fail without the fix?

Yes, I tried it without the fix and they failed with the expected error messages.

> does the readme need updating?

I don’t think so. Was there anything in particular you were thinking about? Either way, I’ll update the readme PR (#473) after this is merged to fix conflicts and see if there’s any more syncing to do.

**Comment by harrysarson** [comment]
I was thinking that we embed the help output in the readme. However, that is probably only true in #473.

**Pull request merged**